File: frontend/iOS/Ride Track/Ride Track/Model/Bike.swift
//
//  Bike.swift
//  Ride Track
//
//  Created by Ben Alexander on 10/28/22.
//

import Foundation

class Bike {
    var id: Int = 0
    var weight: Double = 0
    var name: String = ""
    var user_id: String = ""
    
    init(id: Int, weight: Double, name: String, user_id: String) {
        self.id = id
        self.weight = weight
        self.name = name
        self.user_id = user_id
    }
    
    init() {}
}

File: backend/ride-track-nodejs/src/domain/bike/bike.ts
export class Bike {
    public name: string;
    public weight: number;
    public user_id: string;
    public is_deleted: boolean;
    public id?: string;

    constructor(name: string, weight: number, user_id: string, is_deleted: boolean, id?: string) {
        this.name = name;
        this.weight = weight;
        this.user_id = user_id;
        this.is_deleted = is_deleted;
        this.id = id;
    }
}